packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -464,7 +464,7 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
           .subscribe({
             next: (response) => {
               if (response.status === 201) {
-                if (config.showServiceQueueFields) {
+                if (config.showServiceQueueFields && queueLocation && service && priority) {
```

Comment:
Shouldn't we extend the validation schema as well so that if the user fails to select values for the given fields in the form, they get shown a validation error message and the form fails to submit? Is it mandatory to fill those fields when `showServiceQueueFields` is set to `true`?

Reply:
That would entail extending the schema like so, for example: 

```tsx
  const visitFormSchema = useMemo(() => {
    const visitAttributes = (config.visitAttributeTypes ?? [])?.reduce(
      (acc, { uuid, required }) => ({
        ...acc,
        [uuid]: required
          ? z
              .string({
                required_error: t('fieldRequired', 'This field is required'),
              })
              .refine((value) => !!value, t('fieldRequired', 'This field is required'))
          : z.string().optional(),
      }),
      {},
    );

    const baseSchema = {
      visitStartDate: z.date().refine(
        (value) => {
          const today = dayjs();
          const startDate = dayjs(value);
          return displayVisitStopDateTimeFields ? true : startDate.isSameOrBefore(today, 'day');
        },
        t('invalidVisitStartDate', 'Start date needs to be on or before {{firstEncounterDatetime}}', {
          firstEncounterDatetime: formatDatetime(new Date()),
          interpolation: {
            escapeValue: false,
          },
        }),
      ),
      visitStartTime: z
        .string()
        .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format')),
      visitStartTimeFormat: z.enum(['PM', 'AM']),
      visitStopDate: displayVisitStopDateTimeFields ? z.date() : z.date().optional(),
      visitStopTime: displayVisitStopDateTimeFields
        ? z
            .string()
            .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format'))
        : z.string().optional(),
      visitStopTimeFormat: displayVisitStopDateTimeFields ? z.enum(['PM', 'AM']) : z.enum(['PM', 'AM']).optional(),
      programType: z.string().optional(),
      visitType: z.string().refine((value) => !!value, t('visitTypeRequired', 'Visit type is required')),
      visitLocation: z.object({
        display: z.string(),
        uuid: z.string(),
      }),
      visitAttributes: z.object(visitAttributes),
    };

    return z.object(config.showServiceQueueFields ? { ...baseSchema, ...serviceQueueFieldsSchema } : baseSchema);
  }, [
    config.visitAttributeTypes,
    config.showServiceQueueFields,
    t,
    displayVisitStopDateTimeFields,
    serviceQueueFieldsSchema,
  ]);
```

And then we could pass the schema down into the slot as a piece of state:

```tsx
{config.showServiceQueueFields && (
              <section>
                <div className={styles.sectionTitle}></div>
                <div className={styles.sectionField}>
                  <ExtensionSlot
                    name="visit-form-queue-slot"
                    state={{ setFormFields: setVisitFormFields, schema: serviceQueueFieldsSchema }}
                  />
                </div>
              </section>
            )}
```

And then we'd need to modify the [VisitFormQueueFields component](https://github.com/openmrs/openmrs-esm-patient-management/blob/3a0f42beaf8972d1d060473d8bf888f829ca789f/packages/esm-service-queues-app/src/patient-search/visit-form-queue-fields/visit-form-queue-fields.component.tsx) to leverage the RHF and the zod schema prop.

Reply:
All of this is not strictly necessary for now, though, which is why I proposed addressing it in a subsequent ticket.

Reply:
Thanks @denniskigen I agree we can do this in a follow up PR. @ojwanganto  Whenever queue fields are configured to show on visit form, would it be wise to make it mandatory for them to fill the queue fields or there are exceptional cases?

Reply:
I may not be objective in this. Let's engage the BAs further for a correct and practical position

Reply:
We should rethink form extensibility to avoid hacky solutions like those we've used for billing & queues. Adding features like skip logic and the ability to hide/show fields would help make the forms more flexible and extandable for various workflows. Thoughts?

cc: @denniskigen @ibacher @vasharma05

Reply:
I think, ultimately, we should redesign these forms to actually use the form engine. In fact, I thought there was a way of doing that? Fundamentally, hard-coded forms in an EMR are always going to be a pain-point.